tests/rptest/tests/debug_bundle_test.py

```diff
@@ -73,6 +73,15 @@ def get_metrics_from_node_sync(patterns: list[str]):
             self.metrics), f"Missing expected metrics from node {node.name}"
         return samples
 
+    def _assert_http_error(self, expected_status_code, request, *args,
+                           **kwargs):
+        try:
+            request(*args, **kwargs)
+            assert False, f"Expected HTTPError with status code {expected_status_code}"
+        except requests.HTTPError as e:
+            assert e.response.status_code == expected_status_code, e.response.json(
+            )
```

Comment:
q: is there a reason to prefer this over `expect_exception`?

Reply:
> q: is there a reason to prefer this over `expect_exception`?

Mostly the logging of the request body for debugging if it goes wrong, but maybe I can do that with `expect_exception`?

Reply:
eh, not really. could extend it a bit, but it might be just as wordy.